### PR TITLE
fix: Converts `replication_specs` to TypeList for the advanced_cluster data_source

### PR DESCRIPTION
**Pull request by marcosuma**
## Description

Thanks to @drcapulet for reporting this bug.
1. We are changing `replication_specs` to TypeList in the data_sources because the Set data structure was not showing all the results (due to key clashing most likely)
2. We opted for this option because it's in line with how the resource structure is. So now all `replication_specs` in `advanced_cluster` have TypeList.

Link to any related issue(s):

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Comment by marcosuma** [comment]
Taken from @drcapulet's [PR](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2076/files). Rationale of creating this PR is that we don't allow the whole testing GH action to be fully run on external contributors (and in this case we need it)

**Review by lantoli — APPROVED** [review]

**Review by oarbusi — APPROVED** [review]
LGTM! Thanks @drcapulet for the contribution!

File: .changelog/2145.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+```release-note:bug
+data-source/mongodbatlas_advanced_cluster: Converts `replication_specs` to TypeList
+data-source/mongodbatlas_advanced_clusters: Converts `replication_specs` to TypeList
```
Comment (JuliaMongo) [inline .changelog/2145.txt]: Please add: "This fixes an issue in the `advanced_cluster` data source where `replication_specs` was not returned in the results".  @marcosuma -- here is the review, taken from [your discussion with customer in GH](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2076#issuecomment-2050998555).

**Review by JuliaMongo — APPROVED** [review]
Made one suggestion for the changelog, to make it clear which issue is being addressed for the users, with this fix. Thank you, @marcosuma  for pointing out the change log.

File: .changelog/2145.txt (hunk 1 above)
Comment (marcosuma) [reply .changelog/2145.txt]: Thanks @JuliaMongo, used your input and changed a little bit.

**Pull request merged**